### DISCUSSION TITLE
android sign key not found.

### DISCUSSION THREAD

**Discussion by norbubb** [codemagic-ci-cd/codemagic-docs#2340] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

```
* What went wrong:
Execution failed for task ':app:validateSigningRelease'.
> Keystore file '/Users/builder/clone/android/android_key/upload-keystore.jks' not found for signing config 'release'.
```

### Expected results

sign success

### Actual results

key not found.

### Build id (optional)

65b79bd464df62bf3dced67b

**Comment by dtrdic** [comment]
Hello, @norbubb can you please check if your build.gradle has been configured according to the release signing:

```
signingConfigs{
  release{
    keyAlias System.getenv('CM_KEY_ALIAS')
    keyPassword System.getenv('CM_KEY_PASSWORD')
    storeFile file(System.getenv('CM_KEYSTORE_PATH'))
    storePassword System.getenv('CM_KEYSTORE_PASSWORD')
 }
}
```
Check out[ this section](https://docs.codemagic.io/yaml-code-signing/signing-android/#signing-android-apps-using-gradle).

**Comment by Adventure-Hunt** [comment]
I think I have a related issue. Mine is building ok for Android, but each time the build workflow tries to publish the app its getting a mismatch on the signing fingerprint. The correct keystore file is attached in the build workflow, and the passwords are set correctly. 

The fingerprint of the uploaded file is different every time, even when I put a deliberately incorrect password in, so it appears that the keystore file I am loading is not being used.

I am very new to this, so its probably something simple and silly.

The project is a download from Flutterflow with no editing so far

**Reply by dtrdic** [reply]
hey @Adventure-Hunt, it's most likely the same issue. Please make sure yourr **android/app/build.gradle** file looks [like this](https://docs.codemagic.io/yaml-code-signing/signing-android/#signing-android-apps-using-gradle) in order to  sign your app properly

**Reply by dtrdic** [reply]
I am closing the thread due to inactivity. Let me know if you have any questions by replying to this thread or opening a new one.

**Comment by dudihisine** [comment]
configure your build like this and it will work

```
create("release") {
            // Check if running on Codemagic (it sets CM_KEYSTORE_PATH env variable)
            val codemagicKeystorePath = System.getenv("CM_KEYSTORE_PATH")
            
            if (codemagicKeystorePath != null) {
                // Codemagic build - use environment variables
                storeFile = file(codemagicKeystorePath)
                storePassword = System.getenv("CM_KEYSTORE_PASSWORD") ?: keystoreProperties["storePassword"] as String
                keyAlias = System.getenv("CM_KEY_ALIAS") ?: keystoreProperties["keyAlias"] as String
                keyPassword = System.getenv("CM_KEY_PASSWORD") ?: keystoreProperties["keyPassword"] as String
            } else if (keystoreProperties.containsKey("keyAlias")) {
                // Local build - use key.properties
                keyAlias = keystoreProperties["keyAlias"] as String
                keyPassword = keystoreProperties["keyPassword"] as String
                storeFile = file("../" + keystoreProperties["storeFile"] as String)
                storePassword = keystoreProperties["storePassword"] as String
            }
        }
```
